nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Discover/SdkProjectDiscovery.cs

```diff
@@ -389,14 +451,72 @@ runtimePackageVersion is null ||
                 ImportedFiles = imported,
                 AdditionalFiles = additional,
             };
-        }).ToImmutableArray();
-        return projectDiscoveryResults;
+            projectDiscoveryResults.Add(projectDiscoveryResult);
+        }
+        return projectDiscoveryResults.ToImmutableArray();
+    }
+
+    private static async Task<Dictionary<string, Dictionary<string, Dictionary<string, string>>>> RebuildPackagesPerProject(
```

Comment:
This is the new behavior; take all `<PackageReference>` elements, create a temp project, resolve transitive dependencies, then re-create the desired restore results.